pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -448,19 +445,34 @@ public void disableBroker() throws PulsarServerException {
      */
     @Override
     public synchronized void doLoadShedding() {
+        if (LoadManagerShared.isUnloadDisabledInLoadShedding(pulsar)) {
+            return;
+        }
+        if (getAvailableBrokers().size() <= 1) {
+            log.warn("Only 1 broker available: no load shedding will be performed");
+            return;
+        }
+        // Remove bundles who have been unloaded for longer than the grace period from the recently unloaded
+        // map.
+        final long timeout = System.currentTimeMillis()
+                - TimeUnit.MINUTES.toMillis(conf.getLoadBalancerSheddingGracePeriodMinutes());
+        final Map<String, Long> recentlyUnloadedBundles = loadData.getRecentlyUnloadedBundles();
+        recentlyUnloadedBundles.keySet().removeIf(e -> recentlyUnloadedBundles.get(e) < timeout);
         for (LoadSheddingStrategy strategy : loadSheddingPipeline) {
             final Map<String, String> bundlesToUnload = strategy.findBundlesForUnloading(loadData, conf);
             if (bundlesToUnload != null && !bundlesToUnload.isEmpty()) {
                 try {
                     for (Map.Entry<String, String> entry : bundlesToUnload.entrySet()) {
-                        final String bundle = entry.getKey();
-                        final String broker = entry.getValue();
-                        adminCache.get(broker).namespaces().unloadNamespaceBundle(
+                        final String broker = entry.getKey();
+                        final String bundle = entry.getValue();
+                        log.info("Unloading bundle: {}", bundle);
+                        adminCache.get("http://" + broker).namespaces().unloadNamespaceBundle(
```

Comment:
That reasoning seems sound, I kept the admin cache because that's how it was done on `SimpleLoadManagerImpl`, but since they would not seem to be repercussions for just using one, I will take it out.